### PR TITLE
Ticket #285 | Variable Substitution Depth issue

### PR DESCRIPTION
**Pull request by chawlaasumit**

### CODE REVIEW

**Comment by chtyim** [comment]
What is the JIRA number for this?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java [hunk 1]
```diff
@@ -177,13 +179,14 @@ WorkflowProgramInfo getWorkflowProgramInfo() {
 
   private void setPlugins(Map<String, Plugin> plugins) {
     hConf.set(HCONF_ATTR_PLUGINS, GSON.toJson(plugins, PLUGIN_MAP_TYPE));
+    cConf.set(HCONF_ATTR_PLUGINS, GSON.toJson(plugins, PLUGIN_MAP_TYPE));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java:182]: Why need to store in both cConf and hConf?

Comment (chtyim) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java:182]: The right fix should be not doing variable substitution when getting the plugin attributes from the configuration.

**Review by chtyim — CHANGES_REQUESTED** [review]

**Comment by chawlaasumit** [comment]
creating fresh pull request

**Pull request closed without merging**